sdk/core/core-client/src/utils.ts

```diff
@@ -37,6 +37,59 @@ export function isValidUuid(uuid: string): boolean {
   return validUuidRegex.test(uuid);
 }
 
+/**
+ * Representation of parsed response headers and body coupled with information
+ * about how to map them:
+ * - whether the response body should be wrapped (typically if its type is primitive).
+ * - whether the response is nullable so it can be null if the combination of
+ *   the headers and the body is empty.
+ */
+interface ResponseObjectWithMetadata {
+  /** whether the mapper allows nullable body */
+  hasNullableType: boolean;
```

Comment:
nit: why not reuse `nullable`

Reply:
I think it is good to be concise but in this case `nullable` could be confusing. I think `hasNullableType` is more explicit and clear.